packages/flutter_tools/lib/src/device.dart

```diff
@@ -428,7 +435,7 @@ abstract class PollingDeviceDiscovery extends DeviceDiscovery {
 
   Future<List<Device>> pollingGetDevices({ Duration timeout });
 
-  Future<void> startPolling() async {
+  void startPolling() {
```

Comment:
Most of these methods do not need to be async - this made it much easier to fix the disabled test